internal/core/docker_host.go

```diff
@@ -202,7 +204,10 @@ func extractDockerSocketFromClient(ctx context.Context, cli client.APIClient) st
 		return DockerSocketPath
 	}
 
-	dockerHost := extractDockerHost(ctx)
+	dockerHost, err := extractDockerHost(ctx)
+	if err != nil {
+		panic(err) // Docker host is required to get the Docker socket
```

Comment:
bug: This results in `ExtractDockerSocket` panic which is used in other non internal functions including its namesake `ExtractDockerSocket` none of which mention they can panic.

Reply:
I added comments to the functions, could you please check that? 🙏